csharp/src/Apache.Arrow/Ipc/ArrowReaderImplementation.cs

```diff
@@ -209,4 +243,30 @@ private ArrowBuffer BuildArrowBuffer(ByteBuffer bodyData, Flatbuf.Buffer buffer)
             return new ArrowBuffer(data);
         }
     }
+
+    internal class RecordBatchEnumerator
```

Comment:
(minor perf question) - how do you feel about defining this as a `struct` and passing it `by ref` to each method? That would cut down on this allocation and copying the `Flatbuf.RecordBatch` struct to the heap.

It may not make major perf differences, but it would be less allocations.

Reply:
Sounds good. I'll modify it.